lib/proxy.js

```diff
@@ -137,12 +137,13 @@ function buildResultingMiddleware (host, options, stream) {
     var reqOpts = {
       url     :  URL.format(proxyUrl),
       method  : req.method,
-      headers : _.defaults(options.headers || {}, req.headers),
+      // headers : _.defaults(options.headers || {}, req.headers),
+      headers : options.headers,
```

Comment:
wait I'm not sure what youre doing here? I want the user to be able to explicitly set certain headers but preserve the rest sent by the client. I feel like thats a much more likely need than overwriting all client headers (cookies for example)